src/Common/src/Interop/Unix/System.Security.Cryptography.Native/Interop.Ssl.cs

```diff
@@ -47,7 +47,10 @@ internal static partial class Ssl
 
         [DllImport(Libraries.CryptoNative, EntryPoint = "CryptoNative_SslGetVersion")]
         private static extern IntPtr SslGetVersion(SafeSslHandle ssl);
-                
+
+        [DllImport(Libraries.CryptoNative, EntryPoint = "CryptoNative_SslSetTlsExtHostName")]
+        internal static extern int SslSetTlsExtHostName(SafeSslHandle ssl, string host);
```

Comment:
With no explicit string conversion here, it will be assumed as ASCII but the SNI spec says UTF8

>"HostName" contains the fully qualified DNS hostname of the server,
   as understood by the client. The hostname is represented as a byte
   string using UTF-8 encoding [UTF8], without a trailing dot.

Reply:
FYI its a little bit more complex than that, I don't know what happens on the windows side. But here we have a Unicode string so in theory it should be converted using PunyCode, otherwise if you have a domain name with Chinese unicode chars for instance it won't work, but if it breaks today on windows then maybe that is okay?

Reply:
This part of the spec has changed over the years.
https://tools.ietf.org/html/rfc6066
>   "HostName" contains the fully qualified DNS hostname of the server,
   as understood by the client.  The hostname is represented as a byte
   string using ASCII encoding without a trailing dot.  This allows the
   support of internationalized domain names through the use of A-labels
   defined in [RFC5890].

And by A-labels it means puyncode.

Reply:
LGTM 👍